editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,378 @@
+              switchVersion(defaultVersionName);
+            }
+          } catch (InterruptedException | ExecutionException e) {
+            showAlert("Error applying edits on server");
+          }
+        });
+      }
+    }
+  }
+
+  /**
+   * Switch the active branch version.
+   *
+   * @param versionName name of the version to switch to
+   */
+  private void switchVersion(String versionName) {
+    ListenableFuture<Void> switchVersionFuture = serviceGeodatabase.switchVersionAsync(versionName);
+    switchVersionFuture.addDoneListener(() -> {
+      // check if the active version has switched successfully and update the UI
+      if (serviceGeodatabase.getVersionName().equals(versionName)) {
+        currentVersionLabel.setText("Current version: " + serviceGeodatabase.getVersionName());
+        editFeatureVBox.setDisable(true);
+      } else {
+        showAlert("Error switching version");
+      }
+    });
+  }
+
+  /**
+   * Select a feature if one exists where the user clicked.
+   *
+   * @param point location where the user clicked
+   */
+  private void selectFeature(Point2D point) {
+    ListenableFuture<IdentifyLayerResult> identifyLayerResultFuture = mapView.identifyLayerAsync(featureLayer, point, 1, false);
+    identifyLayerResultFuture.addDoneListener(() -> {
+      try {
+        IdentifyLayerResult identifyLayerResult = identifyLayerResultFuture.get();
+        List<GeoElement> identifiedElements = identifyLayerResult.getElements();
+        if (!identifiedElements.isEmpty()) {
+          GeoElement element = identifiedElements.get(0);
+          if (element instanceof ArcGISFeature) {
+
+            // get the selected feature
+            selectedFeature = (ArcGISFeature) element;
+            featureLayer.selectFeature(selectedFeature);
+            selectedFeature.addDoneLoadingListener(() -> {
+              if (selectedFeature.getLoadStatus() == LoadStatus.LOADED) {
+
+                // when a feature has been selected, get it's damage type value and select it in the damage combo box
+                String selectedFeatureAttributeValue = (String) selectedFeature.getAttributes().get("TYPDAMAGE");
+
+                if (damageTypeComboBox.getItems().contains(selectedFeatureAttributeValue)) {
+                  damageTypeComboBox.getSelectionModel().select(selectedFeatureAttributeValue);
+                } else {
+                  showAlert("Unexpected attribute value");
+                }
+
+                // enable feature editing UI if not on the default version
+                if (!serviceGeodatabase.getVersionName().equals(defaultVersionName)) {
+                  editFeatureVBox.setDisable(false);
+                }
+              } else {
+                showAlert("Feature failed to load");
+              }
+            });
+          }
+        }
+      } catch (InterruptedException | ExecutionException e) {
+        showAlert("Failed to identify the feature");
```

Comment:
another example of where we can improve the alert by providing a meaningful message

```suggestion
        showAlert("Failed to identify the feature" + e.getCause().getMessage());
```